src/Flysystem/AssetAdapter.php

```diff
@@ -146,17 +146,28 @@ protected function configureServer($forceOverwrite = false)
 
         // Apply each configuration
         $config = new FlysystemConfig();
-        $config->set('visibility', $visibility);
         foreach ($configurations as $file => $template) {
             // Ensure file contents
             if ($forceOverwrite || !$this->has($file)) {
+                if ($this->has($file)) {
+                    $perms = $this->getVisibility($file);
+                    if ($perms['visibility'] === $visibility) {
+                        // If the visibility is already correct, do not try to re-set it.
+                        // Attempting to set the visibility of a file/folder when the filesystem is already
+                        // set correctly causes "Operation not permitted" errors on some Linux servers.
+                        $config->set('visibility', null);
+                    } else {
+                        $config->set('visibility', $visibility);
+                    }
```

Comment:
```suggestion
                    // If the visibility is already correct, do not try to re-set it.
                    // Attempting to set the visibility of a file/folder when the filesystem is already
                    // set correctly causes "Operation not permitted" errors on some Linux servers.
                    if ($perms['visibility'] !== $visibility) {
                        $config->set('visibility', $visibility);
                    }
```
I've tested locally that this will fix the behat issue I found on silverstripe/installer

Reply:
Done :)